### PR TITLE
Improve file sync service

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
- file_sync: remove the unused symbols
- file_sync: return 0 when state_process_send_header receive ID_DONE
- file_sync: avoid memory copy in state_process_send_file

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
BTW, @spiriou do you plan migrate file sync service to libuv?

**Comment by spiriou** [comment]
> BTW, @spiriou do you plan migrate file sync service to libuv?

No. I thought about handling non-blocking IOs for FS but it adds a lot of complexity so I stopped.

I only plan to finish `adb reverse/forward` implementation for now.

**Comment by xiaoxiang781216** [comment]
> > BTW, @spiriou do you plan migrate file sync service to libuv?
> 
> No. I thought about handling non-blocking IOs for FS but it adds a lot of complexity so I stopped.
> 
> I only plan to finish `adb reverse/forward` implementation for now.

Nice, when you will finish the work?

**Comment by spiriou** [comment]
> > > BTW, @spiriou do you plan migrate file sync service to libuv?
> > 
> > 
> > No. I thought about handling non-blocking IOs for FS but it adds a lot of complexity so I stopped.
> > I only plan to finish `adb reverse/forward` implementation for now.
> 
> Nice, when you will finish the work?

Soon. The feature is in the `draft` branch but I was not happy with the API and libuv abstraction so it's still not in master.

File: file_sync_service.c [hunk 1]
```diff
@@ -534,45 +534,37 @@ static int state_process_send_header(afs_service_t *svc, apacket *p)
...
-        ret = read_from_packet(svc, p, block_size);
-        if (ret != 0) {
-            if (ret == -EAGAIN) {
-                return 1;
+    int block_size = min(p->msg.data_length, svc->namelen);
+    uint8_t *write_ptr = svc->packet_ptr;
+
+    svc->namelen -= block_size;
+    p->msg.data_length -= block_size;
+    svc->packet_ptr += block_size;
```
Comment (spiriou) [inline file_sync_service.c:542]: IIUC I see two changes here:
- `while` loop is removed to rely on the caller's one
- `read_from_packet()` is removed to optimize the `write()` without going through the `memcpy()`. For this change I would add another function or macro to handle in the same place the `packet_ptr` and `p->msgdata_length` logic. I'll merge this PR and think about this.

Comment (xiaoxiang781216) [reply file_sync_service.c:542]: Sure.

**Comment by xiaoxiang781216** [comment]
> > > > BTW, @spiriou do you plan migrate file sync service to libuv?
> > > 
> > > 
> > > No. I thought about handling non-blocking IOs for FS but it adds a lot of complexity so I stopped.
> > > I only plan to finish `adb reverse/forward` implementation for now.
> > 
> > 
> > Nice, when you will finish the work?
> 
> Soon. The feature is in the `draft` branch but I was not happy with the API and libuv abstraction so it's still not in master.

Good, I will take a look.

**Pull request merged**